### DISCUSSION TITLE
How to connect to a public session using lua?

### DISCUSSION THREAD

**Discussion by ghost** [YimMenu/YimMenu#1869] [Q&A]
I just don't want to do it manually, but using a script

**Comment by maybegreat48** [comment]
Connect by Rockstar ID?

**Reply by Yimura** [reply]
Exposing this through a command would be a good start most likely.

https://github.com/YimMenu/YimMenu/blob/96493b5a22d9f57c884274797dcb2e23d338851c/src/util/session.hpp#L38-L80

**Comment by ghost** [comment]
> Connect by Rockstar ID?

No, I meant this is how it is implemented in the session switcher, but only through the command. But the ability to implement a connection via RID will also not be superfluous

**Comment by ghost** [comment]
> It is implemented through globals, Use the following
> 
> 1574589 on call set to 1
> 
> 1575020 set the session type to join, -1 pub sessions etc, all in yimmenu somewhere (take a look at the session joiner)

Sorry, I wanted to ask again. If I try to join a public session, I see a non-disappearing "Multiplayer Swoop Up" label. Maybe I did something wrong?
`
globals.set_int(1574589, 1)
globals.set_int(1575020, -1)
`
![pic](https://github.com/YimMenu/YimMenu/assets/139077095/6fcb4830-e759-48d1-8d30-835ed74f820f)

**Reply by Yimura** [reply]
https://github.com/YimMenu/YimMenu/blob/master/src/util/session.hpp#L47-L55

This is the exact code to just do the session switching. You might have switched around the two globals.

And these are the values to switch to specific session types.
https://github.com/YimMenu/YimMenu/blob/master/src/core/enums.hpp#L216-L229

**Comment by ghost** [comment]
> https://github.com/YimMenu/YimMenu/blob/master/src/util/session.hpp#L47-L55
> 
> This is the exact code to just do the session switching. You might have switched around the two globals.
> 
> And these are the values to switch to specific session types. https://github.com/YimMenu/YimMenu/blob/master/src/core/enums.hpp#L216-L229

Damn, as soon as I did not rearrange the two globals anyway, I'm in the same session and I see "Multiplayer swoops up". I just don't know what I'm doing wrong. I tried to do so as in the session.hpp file:
   `globals.set_int(1574589, -1)
    globals.set_int(1574589, 0)
    globals.set_int(1575020, 0)
    globals.set_int(1574589, 1)` And still I stay in the same session and see the same label.

**Reply by ghost** [reply]
i used the button but it still doesn't work
